File: python/src/xiaoyi/domain/token/syntax/primitive/int8/kind.py
"""Syntax primitive integer token kind: INT8.

Layer hierarchy:
- 1 syntax
- 2 primitive
- 3 int8
- 4 kind

Concrete variant taxonomy for INT8 syntax nodes before rendering.
"""

from __future__ import annotations

from enum import StrEnum


class Int8Kind(StrEnum):
    """Classification of INT8 token kinds."""

    LITERAL = "INT8_LITERAL"
    VARIABLE = "INT8_VAR"
    CAST = "INT8_CAST"

    def label(self) -> str:
        """Return the token label for parser diagnostics."""
        return self.value

File: ts/src/xiaoyi/domain/token/syntax/primitive/int8/kind.ts
/**
 * Syntax primitive integer token kind: INT8.
 *
 * Layer hierarchy:
 * - 1 syntax
 * - 2 primitive
 * - 3 int8
 * - 4 kind
 *
 * Concrete variant taxonomy for INT8 syntax nodes before rendering.
 */

export type Int8Kind = 'INT8_LITERAL' | 'INT8_VAR' | 'INT8_CAST';

export const INT8_KIND: readonly Int8Kind[] = [
  'INT8_LITERAL',
  'INT8_VAR',
  'INT8_CAST',
];

export function labelInt8Kind(kind: Int8Kind): string {
  return kind;
}
